packages/sdk/src/evm/functions/contract/index.ts

```diff
@@ -0,0 +1,102 @@
+import { getCompositePluginABI } from "../../common/plugin";
+import { PREBUILT_CONTRACTS_MAP } from "../../contracts";
+import { SmartContract } from "../../contracts/smart-contract";
+import { ContractPublisher } from "../../core/classes/contract-publisher";
+import { getSignerAndProvider } from "../../core/classes/rpc-connection-handler";
+import { ValidContractInstance } from "../../core/types";
+import { AbiSchema } from "../../schema/contracts/custom";
+import {
+  cacheContract,
+  getCachedContract,
+  getCachedStorage,
+  inContractCache,
+} from "../core/cache";
+import { GetContractFromAbiParams, GetContractParams } from "./types";
+import { resolveContractType } from "./utils";
+
+export async function getContractFromAbi(
+  params: GetContractFromAbiParams,
+): Promise<ValidContractInstance> {
+  if (inContractCache(params.address)) {
+    return getCachedContract(params.address);
+  }
+
+  // TODO: Remove support for SDKOptions
+  const [signer, provider] = getSignerAndProvider(
+    params.network,
+    params.sdkOptions,
+  );
+
+  const parsedAbi =
+    typeof params.abi === "string" ? JSON.parse(params.abi) : params.abi;
+  const contract = new SmartContract(
+    signer || provider,
+    params.address,
+    await getCompositePluginABI(
+      params.address,
+      AbiSchema.parse(parsedAbi),
+      provider,
+      params.sdkOptions,
+      getCachedStorage(params.storage),
+    ),
+    getCachedStorage(params.storage),
+    params.sdkOptions,
+    (await provider.getNetwork()).chainId,
+  );
+
+  cacheContract(params.address, contract);
+  return contract;
+}
+
+export async function getContract(params: GetContractParams) {
+  if (inContractCache(params.address)) {
+    return getCachedContract(params.address);
+  }
+
+  const [signer, provider] = getSignerAndProvider(
+    params.network,
+    params.sdkOptions,
+  );
+
+  if (!params.contractTypeOrAbi || params.contractTypeOrAbi === "custom") {
+    const contractType = await resolveContractType({
+      address: params.address,
+      provider,
+    });
+    if (contractType === "custom") {
+      // Not even worth cacheing publisher since we would have to cache by network, which requires an RPC call
```

Comment:
why would we not cache publisher by network?

Reply:
globally?

Reply:
b/c checking if we have a cached `ContractPublisher` requires making an RPC request to get the current network we're on, which is an extra request and potentially slower than just instantiating a new one (?).

Reply:
I guess we'll be making that request to cache contract now anyway